src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    const { type } = action;
+    let nextState;
+
+    switch (type) {
+      case 'addProperties':
+        const { extraData } = action;
+
+        nextState = {
```

Comment:
Avoid using an unnecessary comma at the end of the last property. Although it is not a syntax error, it's better to keep the code clean and consistent.